sdk/containerregistry/container-registry/recordings/node/repository_and_artifact_tests/recording_should_retrive_registry_artifact_properties_for_a_tag.json

```diff
@@ -0,0 +1,352 @@
+{
+  "Entries": [
+    {
+      "RequestUri": "https://myregistry.azurecr.io/acr/v1/library%2Fhello-world/_tags/test1?api-version=2021-07-01",
+      "RequestMethod": "GET",
+      "RequestHeaders": {
+        "Accept": "application/json",
+        "Accept-Encoding": "gzip,deflate",
+        "Connection": "keep-alive",
+        "User-Agent": "azsdk-js-container-registry/1.0.1 core-rest-pipeline/1.5.1 Node/v16.13.0 OS/(x64-Linux-5.10.60.1-microsoft-standard-WSL2)",
+        "x-ms-client-request-id": "ed944744-972f-427c-874a-0aceea9dcc05"
+      },
+      "RequestBody": null,
+      "StatusCode": 401,
+      "ResponseHeaders": {
+        "Access-Control-Expose-Headers": [
+          "Docker-Content-Digest",
+          "WWW-Authenticate",
+          "Link",
+          "X-Ms-Correlation-Request-Id"
+        ],
+        "Connection": "keep-alive",
+        "Content-Length": "222",
+        "Content-Type": "application/json; charset=utf-8",
+        "Date": "Thu, 10 Feb 2022 00:02:50 GMT",
+        "Docker-Distribution-Api-Version": "registry/2.0",
+        "Server": "openresty",
+        "Strict-Transport-Security": [
+          "max-age=31536000; includeSubDomains",
+          "max-age=31536000; includeSubDomains"
+        ],
+        "WWW-Authenticate": "Bearer realm=\u0022https://myregistry.azurecr.io/oauth2/token\u0022,service=\u0022myregistry.azurecr.io\u0022,scope=\u0022repository:library/hello-world:metadata_read\u0022",
+        "X-Content-Type-Options": "nosniff",
+        "X-Ms-Correlation-Request-Id": "f823f8b8-71d2-46f3-9e2a-d589505d076e"
+      },
+      "ResponseBody": {
+        "errors": [
+          {
+            "code": "UNAUTHORIZED",
+            "message": "authentication required, visit https://aka.ms/acr/authorization for more information.",
```

Comment:
wait what?

Looks like this unathorized is happing in many tests.
Is that supposed to happen, @jeremymeng ?

Reply:
You'll see that this shows up in the old recordings too, as I understand it it's supposed to happen.

Reply:
Yeah, not blaming your PR 🙂

Looks like something is wrong with the SDK or the service and someone has to look?

Reply:
It's expected. This is what we called challenge-based-authentication. The service returns 401 and a `WWW-Authenticate` header that contains the challenge, ACR SDK handles the challenge by exchange the AAD token for an ACR refresh token below, then use that to get an ACR access token which is then used to access ACR API.